### PR TITLE
tests-ts-tls-passthrough

### PR DESCRIPTION
**Pull request by vepatel**
### Proposed changes
- Add test for tls passthrough
- Add test for ts host collision
- general refactoring

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [ ] I have rebased my branch onto master
- [ ] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: tests/suite/test_ts_tls_passthrough.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        """
+        print("Step 1: Create VirtualServer with same host")
+        vs_src_same_host = (
+            f"{TEST_DATA}/transport-server-tls-passthrough/virtual-server-same-host.yaml"
+        )
+        vs_same_host_name = create_virtual_server_from_yaml(
+            kube_apis.custom_objects, vs_src_same_host, test_namespace
+        )
+        wait_before_test(1)
+        response = read_custom_resource(kube_apis.custom_objects, test_namespace, "virtualservers", vs_same_host_name)
```
Comment (pleshakov) [inline tests/suite/test_ts_tls_passthrough.py:219]: would it make sense to have a helper read_vs function, similar to read_ts?

File: tests/suite/test_ts_tls_passthrough.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+@pytest.mark.ts
+@pytest.mark.parametrize(
+    "crd_ingress_controller, transport_server_tls_passthrough_setup",
+    [
+        (
+            {
+                "type": "complete",
+                "extra_args": [
+                    "-enable-custom-resources",
```
Comment (pleshakov) [inline tests/suite/test_ts_tls_passthrough.py:97]: since `-enable-custom-resources` is true by default, would it make sense to get rid of it here?

**Review by pleshakov — APPROVED** [review]

File: tests/suite/test_ts_tls_passthrough.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+            f"{transport_server_tls_passthrough_setup.public_endpoint.port_ssl}"
+        )
+        wait_before_test()
+        resp = session.get(
+            req_url,
+            headers={"host": transport_server_tls_passthrough_setup.ts_host},
+            verify=False,
+        )
+        assert resp.status_code == 200
+        assert f"hello from pod {get_first_pod_name(kube_apis.v1, test_namespace)}" in resp.text
```
Comment (soneillf5) [inline tests/suite/test_ts_tls_passthrough.py:144]: nice test.

File: tests/suite/test_ts_tls_passthrough.py [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+            and response["status"]["message"] == "Host is taken by another resource"
+        )
+
+        print("Step 2: Delete TS taking up the host")
+        delete_ts(
+            kube_apis.custom_objects,
+            transport_server_tls_passthrough_setup.ts_resource,
+            test_namespace,
+        )
+        wait_before_test(1)
```
Comment (soneillf5) [inline tests/suite/test_ts_tls_passthrough.py:178]: shame about these sleep commands, do we have any `waitUntil(condition, timeout)` style functions available ?

**Review by soneillf5 — COMMENTED** [review]
Minor question around using a `waitUntil` method

**Review by soneillf5 — APPROVED** [review]

**Pull request merged**